### DISCUSSION TITLE
Dledger集群模式下，主从数据没有同步

### DISCUSSION THREAD

**Discussion by xiegongshan** [apache/rocketmq#8307] [Q&A]
![master](https://github.com/apache/rocketmq/assets/41192598/e7d7b4ee-e326-4baa-9361-f10b2d537c5c)
![slave](https://github.com/apache/rocketmq/assets/41192598/230d7eda-87a9-4e86-a3a5-74adf2fe3469)
![slave1](https://github.com/apache/rocketmq/assets/41192598/5e7e1d5b-a036-4b18-b62e-ba94ba08f758)

**Comment by xiegongshan** [comment]
![管理页面](https://github.com/apache/rocketmq/assets/41192598/ffb33c42-1ae5-4a2b-8eb4-7d1d58770ae0)

**Reply by xiegongshan** [reply]
这是一个补充

**Reply by cserwen** [reply]
可以补充下 broker_default.log 中的日志信息

**Reply by xiegongshan** [reply]
> broker_default.

![default_broker日志](https://github.com/apache/rocketmq/assets/41192598/25b12c2a-568c-48c2-9b9d-e016e5fc7e73)
几乎都是一些重复的信息

**Reply by xiegongshan** [reply]
> 可以补充下 broker_default.log 中的日志信息

您好，能在哪里看到同步失败的错误日志呢？

**Reply by cserwen** [reply]
是在正常同步的，可以对比下 各个 Broker 中的存储文件，如果一致就说明同步是正常进行的，可能需要排查下监控的问题 @xiegongshan

**Reply by xiegongshan** [reply]
> Broker 中的存储文件

您好我发现我没有生成commitlog文件。

**Reply by cserwen** [reply]
dledger 不是通过 storePathCommitLog 配置的，你看下 storePathRootDir/dledger-n0/data 这里应该有

**Reply by xiegongshan** [reply]
> dledger 不是通过 storePathCommitLog 配置的，你看下 storePathRootDir/dledger-n0/data 这里应该有
![commit错误](https://github.com/apache/rocketmq/assets/41192598/cf7ecc00-a744-417f-9800-bd30d4d14d8d)

data下面只有一个000000000000000的文件

然后/tmp/rmqstore/node00/commitlog 下面也没有任何内容

**Reply by xiegongshan** [reply]
> dledger 不是通过 storePathCommitLog 配置的，你看下 storePathRootDir/dledger-n0/data 这里应该有

或者您是否能给我一个dledger集群模式的部署文档呢，我参考着重新来一次。

**Comment by wyattzv** [comment]
碰到相同的问题 请问下怎么解决的 @xiegongshan

**Comment by hakusai22** [comment]
hello 想问下你部署的1master 2slave节点 当master down了之后 slave切到master节点 有写消息的能力吗？

**Comment by AliBaBajavaer** [comment]
关于DLedger,如果某条DLedgerEntry主从同步超时会怎么处理?我看源码只说到了返回写入失败,那master,需要删除这条消息吗
